BUG: loadtxt raises an issue on file that used to work

### Describe the issue:

Thanks to the `--pre` wheels, our [CIs found some issues](https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=18059&view=logs&j=305851a9-a7bb-55db-0042-7e2b6f48aa1c&t=77eb5bf7-3205-5b54-06c6-183c8195d846&l=3053) with (what I assume is) the `loadtxt` change to C.

### Reproduce the code example:

1. Running `np.loadtxt(fname, skiprows=1)` on [this file](https://github.com/mne-tools/mne-testing-data/blob/master/SSS/test_move_anon_raw.pos) or [this file](https://github.com/mne-tools/mne-testing-data/blob/master/SSS/chpi5_raw_mc.pos) both used to work fine, now we get:
    ```
    mne\chpi.py:84: in read_head_pos
        data = np.loadtxt(fname, skiprows=1)  # first line is header, skip it
    C:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\numpy\lib\npyio.py:1321: in loadtxt
        arr = _read(fname, dtype=dtype, comment=comment, delimiter=delimiter,
    C:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\numpy\lib\npyio.py:999: in _read
        arr = _load_from_filelike(
    E   ValueError: could not convert string '' to float64 at row 0, column 11.
    ```

2. Running `with open(egi_txt_fname) as fid: data = np.loadtxt(fid)` on [this file](https://github.com/mne-tools/mne-python/blob/main/mne/io/egi/tests/data/test_egi.txt) used to work fine, now we get:

    ```
    C:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\numpy\lib\npyio.py:1321: in loadtxt
        arr = _read(fname, dtype=dtype, comment=comment, delimiter=delimiter,
    C:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\numpy\lib\npyio.py:999: in _read
        arr = _load_from_filelike(
    E   ValueError: could not convert string '' to float64 at row 0, column 78.
    ```

### Error message:

_No response_

### NumPy/Python version information:

Platform:            Windows-10-10.0.17763-SP0
Python:              3.9.9 (tags/v3.9.9:ccb0e6a, Nov 15 2021, 18:08:50) [MSC v.1929 64 bit (AMD64)]
numpy:               1.23.0.dev0+735.g4c60b3263

--- a/numpy/core/src/multiarray/textreading/tokenize.cpp
+++ b/numpy/core/src/multiarray/textreading/tokenize.cpp
@@ -378,13 +378,22 @@ tokenize(stream *s, tokenizer_state *ts, parser_config *const config)
     ts->num_fields -= 1;
 
     /*
-     * If have one field, but that field is completely empty, this is an
-     * empty line, and we just ignore it.
+     * We always start a new field (at the very beginning and whenever a
+     * delimiter was found).
+     * This gives us two scenarios where we need to ignore the last field
+     * if it is empty:
+     * 1. If there is exactly one empty (unquoted) field, the whole line is
+     *    empty.
+     * 2. If we are splitting on whitespace we always ignore a last empty
+     *    field to match Python's splitting: `" 1 ".split()`.
      */
     if (ts->num_fields == 1
-             && ts->fields[1].offset - ts->fields[0].offset == 1
-             && !ts->fields->quoted) {
-        ts->num_fields--;
+            || ts->unquoted_state == TOKENIZE_UNQUOTED_WHITESPACE) {
+        ssize_t offset_last = ts->fields[ts->num_fields-1].offset;
+        ssize_t end_last = ts->fields[ts->num_fields].offset;
+        if (!ts->fields->quoted && end_last - offset_last == 1) {
+            ts->num_fields--;
+        }
     }
     ts->state = TOKENIZE_INIT;
     return finished_reading_file;

--- a/numpy/lib/tests/test_loadtxt.py
+++ b/numpy/lib/tests/test_loadtxt.py
@@ -165,6 +165,7 @@ def test_bad_ndmin(badval):
 @pytest.mark.parametrize(
     "ws",
     (
+            " ",  # space
             "\t",  # tab
             "\u2003",  # em
             "\u00A0",  # non-break
@@ -173,7 +174,10 @@ def test_bad_ndmin(badval):
 )
 def test_blank_lines_spaces_delimit(ws):
     txt = StringIO(
-        f"1 2{ws}30\n\n4 5 60\n  {ws}  \n7 8 {ws} 90\n  # comment\n3 2 1"
+        f"1 2{ws}30\n\n{ws}\n"
+        f"4 5 60{ws}\n  {ws}  \n"
+        f"7 8 {ws} 90\n  # comment\n"
+        f"3 2 1"
     )
     # NOTE: It is unclear that the `  # comment` should succeed. Except
     #       for delimiter=None, which should use any whitespace (and maybe

